python/kfserving/kfserving/storage.py

```diff
@@ -43,15 +47,59 @@ def download(uri: str) -> str:
 
     @staticmethod
     def _download_s3(uri, temp_dir: str):
-        raise NotImplementedError
+        client = Storage._create_minio_client()
+        try:
+            bucket_args = uri.replace(_S3_PREFIX, "", 1).split("/", 1)
+            bucket_name = bucket_args[0]
+            bucket_path = bucket_args[1] if len(bucket_args) > 1 else ""
+            objects = client.list_objects(bucket_name, prefix=bucket_path, recursive=True)
+            for obj in objects:
+                object_file_name = obj.object_name.replace(bucket_path, "", 1).strip("/")
+                client.fget_object(bucket_name, obj.object_name, temp_dir + "/" + object_file_name)
+        except error.NoSuchBucket as e:
+            raise error.NoSuchBucket("Bucket is not found, please double check the bucket name.")
```

Comment:
Can we just pass the error directly?

Reply:
I updated the PR now to pass errors unless it's necessary to catch it.